src/main/java/com/google/firebase/iid/FirebaseInstanceId.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.iid;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.client.http.HttpResponseInterceptor;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.JsonObjectParser;
+import com.google.api.core.ApiFuture;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.io.ByteStreams;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.ImplFirebaseTrampolines;
+import com.google.firebase.internal.FirebaseService;
+import com.google.firebase.internal.NonNull;
+import com.google.firebase.internal.TaskToApiFuture;
+import com.google.firebase.tasks.Task;
+
+import java.util.concurrent.Callable;
+
+/**
+ * This class is the entry point for all server-side Firebase Instance ID actions.
+ *
+ * <p>Enables deleting instance IDs associated with Firebase projects.
+ */
+public class FirebaseInstanceId {
+
+  private static final String IID_SERVICE_URL = "https://console.firebase.google.com/v1";
+
+  private final FirebaseApp app;
+  private final HttpRequestFactory requestFactory;
+  private final JsonFactory jsonFactory;
+  private final String projectId;
+
+  private HttpResponseInterceptor interceptor;
+
+  private FirebaseInstanceId(FirebaseApp app) {
+    HttpTransport httpTransport = app.getOptions().getHttpTransport();
+    GoogleCredentials credentials = ImplFirebaseTrampolines.getCredentials(app);
+    this.app = app;
+    this.requestFactory = httpTransport.createRequestFactory(
+        new HttpCredentialsAdapter(credentials));
+    this.jsonFactory = app.getOptions().getJsonFactory();
+    this.projectId = ImplFirebaseTrampolines.getProjectId(app);
+    checkArgument(!Strings.isNullOrEmpty(projectId),
+        "Project ID is required to access instance ID service. Use a service account credential or "
```

Comment:
Does it make sense to define this message globally? I assume we throw it in many places.

Reply:
So far we got 2 -- Firestore and IID. I will refactor it out if a 3rd one comes along. It's a little weird, since we also customize the message a little bit for each service. But we should be able to move this into `ImplFirebaseTrampolines` itself.